src/praisonai-agents/praisonaiagents/agent/execution_mixin.py

```
@@ -297,7 +297,28 @@ async def _astart_with_outcome(self, prompt, timeout=None, **kwargs):
+        recorded a provider block/refusal/truncation on ``last_stop_reason`` from
+        the LLM ``finish_reason``/refusal signal — in which case the specific,
+        actionable terminal reason (``content_filtered | refused |
+        length_truncated``) is surfaced instead of a silent empty ``completed``.
+        Any other stop reason (``completed``/``max_steps``/unknown) preserves the
+        existing ``completed`` semantics, so behaviour is unchanged on success.
+        """
+        from .run_outcome import RunOutcome, PROVIDER_BLOCK_REASONS
+        output = str(result) if result is not None else None
+        try:
+            reason = getattr(self, "last_stop_reason", None)
+        except Exception:
```

Ruff (0.16.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)